### PR TITLE
[Merged by Bors] - Thread local example cleanup

### LINKED ISSUE #1917: ECS Guide exclusive system (thread local system) is not useful because of dead code
## How can Bevy's documentation be improved?

ECS Guide exclusive system (thread local system) is not useful because of dead code:
https://github.com/bevyengine/bevy/blob/v0.5.0/examples/ecs/ecs_guide.rs#L203
It has example of thread local system, but no add_system call in app builder.

It was probably 2 times i had to look in examples to find what was the syntax for thread local system (when it was bevy 0.3/4). Now when they are renamed to exclusive access i had to look once again to see new registration syntax. I found it in scenes.
I also found no usage of exclusive/thread local system in any example in `examples/ecs`.

I suggest to either make it actual part of ecs guide or extract it into additional example in ecs folder.

### PR DESCRIPTION
**Pull request by dbearden**
# Objective

Fixes #1917

## Solution

- Change use of "thread local system" wording to "exclusive system".
- Add .exclusive_system() call to example.

### CODE REVIEW

File: examples/ecs/ecs_guide.rs [hunk 1]
```diff
@@ -329,6 +329,14 @@ fn main() {
         )
         .add_system_to_stage(MyStage::BeforeRound, new_round_system)
         .add_system_to_stage(MyStage::BeforeRound, new_player_system)
+        .add_system_to_stage(
+            MyStage::BeforeRound,
+            exclusive_player_system.exclusive_system(),
+        )
+        // Systems which take a `World` as an argument must call `.exclusive_system()`.
```
Comment (alice-i-cecile) [inline examples/ecs/ecs_guide.rs:336]:
```suggestion
        // Systems which take `&mut World` as an argument must call `.exclusive_system()`.
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
One tiny nit, then I'm happy with this! It's good to get the outdated language cleaned up here finally.

**Review by alice-i-cecile — APPROVED** [review]

**Review by killercup — APPROVED** [review]

File: examples/ecs/ecs_guide.rs [hunk 2]
```diff
@@ -198,12 +198,12 @@ fn new_player_system(
...
 }
 
-// If you really need full, immediate read/write access to the world or resources, you can use a
-// "thread local system". These run on the main app thread (hence the name "thread local")
+// If you really need full, immediate read/write access to the world or resources, you can use an
+// "exclusive system".
 // WARNING: These will block all parallel execution of other systems until they finish, so they
-// should generally be avoided if you care about performance
+// should generally be avoided if you care about performance.
 #[allow(dead_code)]
```
Comment (mockersf) [inline examples/ecs/ecs_guide.rs:205]: I think this is not dead code anymore, this annotation should be removed

**Review by mockersf — CHANGES_REQUESTED** [review]

**Pull request closed without merging**